test/sjcl-ecdsa-signdeterministic-test.js

```diff
@@ -0,0 +1,153 @@
+      var secret_key = new sjcl.ecc.ecdsa.secretKey(curve, secret_bn);
+      var hashed_message = hash_function.hash(message);
+
+      var signature = secret_key.signDeterministic(
+        hashed_message,
+        hash_function
+      );
+
+      var pub_val_point = secret_key._curve.G.mult(secret_key._exponent);
+      var public_key = new sjcl.ecc.ecdsa.publicKey(curve, pub_val_point);
+
+      assert(public_key.verify(hashed_message, signature));
+    });
+
+  });
+
+  var rfcCurve = sjcl.ecc.curves.nist_p256;
+  var rfcQ = sjcl.bn.fromBits(sjcl.codec.hex.toBits(
+    'FFFFFFFF00000000FFFFFFFFFFFFFFFFBCE6FAADA7179E84F3B9CAC2FC632551'
+  ));
+  var rfcPrivate = new sjcl.ecc.ecdsa.secretKey(
+    rfcCurve,
+    sjcl.bn.fromBits(sjcl.codec.hex.toBits(
+      'C9AFA9D845BA75166B5C215767B1D6934E50C3DB36E89B127B8A622B120F6721'
+  )));
+  var rfcPublic = new sjcl.ecc.ecdsa.publicKey(
+    rfcCurve,
+    (rfcPrivate._curve.G.mult(rfcPrivate._exponent))
+  );
+  var rfcTests = [
+    { 
+      hash: sjcl.hash.sha1,
+      message: 'sample',
+      k: '882905f1227fd620fbf2abf21244f0ba83d0dc3a9103dbbee43a1fb858109db4',
+      r: '61340c88c3aaebeb4f6d667f672ca9759a6ccaa9fa8811313039ee4a35471d32',
+      s: '6d7f147dac089441bb2e2fe8f7a3fa264b9c475098fdcf6e00d7c996e1b8b7eb'
+    }, /*{ 
+      hash: sjcl.hash.sha224,
+      message: 'sample',
+      k: '103f90ee9dc52e5e7fb5132b7033c63066d194321491862059967c715985d473',
+      r: '53b2fff5d1752b2c689df257c04c40a587fababb3f6fc2702f1343af7ca9aa3f',
+      s: 'b9afb64fdc03dc1a131c7d2386d11e349f070aa432a4acc918bea988bf75c74c'
+    },*/ { 
+      hash: sjcl.hash.sha256,
+      message: 'sample',
+      k: 'a6e3c57dd01abe90086538398355dd4c3b17aa873382b0f24d6129493d8aad60',
+      r: 'efd48b2aacb6a8fd1140dd9cd45e81d69d2c877b56aaf991c34d0ea84eaf3716',
+      s: 'f7cb1c942d657c41d436c7a1b6e29f65f3e900dbb9aff4064dc4ab2f843acda8'
+    },/* { 
+      hash: sjcl.hash.sha384,
+      message: 'sample',
+      k: '09f634b188cefd98e7ec88b1aa9852d734d0bc272f7d2a47decc6ebeb375aad4',
+      r: '0eafea039b20e9b42309fb1d89e213057cbf973dc0cfc8f129edddc800ef7719',
+      s: '4861f0491e6998b9455193e34e7b0d284ddd7149a74b95b9261f13abde940954'
+    },*/ { 
+      hash: sjcl.hash.sha512,
+      message: 'sample',
+      k: '5fa81c63109badb88c1f367b47da606da28cad69aa22c4fe6ad7df73a7173aa5',
+      r: '8496a60b5e9b47c825488827e0495b0e3fa109ec4568fd3f8d1097678eb97f00',
+      s: '2362ab1adbe2b8adf9cb9edab740ea6049c028114f2460f96554f61fae3302fe'
+    }
+  ];
+
+  
+  rfcTests.forEach(function (test, index) {
+
+    it('Should create the right k and s for RFC test #' + index, function() {
+      
+      var hashed_message = test.hash.hash(test.message);
+      var k = rfcPrivate.generateK(hashed_message, test.hash, rfcQ);
+
+      console.log(sjcl.codec.hex.fromBits(k.toBits()));
+      console.log(test.k);
+      assert(sjcl.codec.hex.fromBits(k.toBits()), test.k);
+      var s = rfcPrivate.signDeterministic(hashed_message, test.hash);
```

Comment:
`s` returned here, I think is a bitArray concat of r and s, so you need to destructure it before comparing against `test.s` and `test.r`